python/ipyleaflet/ipyleaflet/leaflet.py

```diff
@@ -1118,13 +1124,25 @@ class VectorTileLayer(Layer):
     url = Unicode().tag(sync=True, o=True)
     attribution = Unicode().tag(sync=True, o=True)
 
-    vector_tile_layer_styles = Union([Dict(), Unicode()]).tag(sync=True, o=True)
-    opacity = Float(1.0, min=0.0, max=1.0).tag(sync=True)
-    visible = Bool(True).tag(sync=True)
+    layer_styles = Union([Dict(), Unicode()]).tag(sync=True, o=True)
+    vector_tile_layer_styles = Union([Dict(), Unicode()], allow_none=True, default_value=None)
+    opacity = Float(1.0, min=0.0, max=1.0).tag(sync=True,o=True)
+    visible = Bool(True).tag(sync=True, o=True)
+    interactive = Bool(False).tag(sync=True, o=True)
     min_zoom = Int(0).tag(sync=True, o=True)
     max_zoom = Int(18).tag(sync=True, o=True)
     min_native_zoom = Int(default_value=None, allow_none=True).tag(sync=True, o=True)
     max_native_zoom = Int(default_value=None, allow_none=True).tag(sync=True, o=True)
+    renderer = Unicode('svg').tag(sync=True, o=True)
+    feature_id = Unicode(allow_none=True, default_value=None).tag(sync=True, o=True)
+    feature_style = Dict().tag(sync=True)
+
+    def __init__(self, **kwargs):
+        super(VectorTileLayer, self).__init__(**kwargs)
+        # Backwards compatibility: allow vector_tile_layer_styles as input:
```

Comment:
This will be backward compatible only at the layer creation.

I see three options here:
- should we really rename this property? If not we can keep `vector_tile_layer_styles`?
If yes:
- we remain backward compatible and we'd need a Python setter too `@vector_tile_layer_styles.setter def set_vector_tile_layer_styles`
- we remove backward compatibility and go with 0.20.0

I'd prefer one of the two first options

Reply:
@martinRenou 

I went ahead with option 2: `vector_tile_layer_styles` is backwards compatible, and it uses a setter that updates `layer_styles`. 

Regarding the ability to update styles after initialization -- this wasn't contemplated in #1206, so I also fixed this by adding a listener to `layer_styles`. 

Here's a demo of the new capabilities: 

<img src="https://github.com/user-attachments/assets/44689670-4961-4583-a1c0-a8882c20c0a0" width="400">